src/mode/jinja_completions.js

```diff
@@ -0,0 +1,89 @@
+
+"use strict";
+
+var TokenIterator = require("../token_iterator").TokenIterator;
+
+var jinjaFilters = [
+  "abs", "float", "lower", "round", "tojson", "attr",
+  "forceescape", "map", "safe", "trim", "batch", "format",
+  "max", "select", "truncate", "capitalize", "groupby", "min",
+  "selectattr", "unique", "center", "indent", "pprint", "slice",
+  "upper", "default", "int", "random", "sort", "urlencode",
+  "dictsort", "join", "reject", "string", "urlize", "escape",
+  "last", "rejectattr", "striptags", "wordcount", "filesizeformat",
+  "length", "replace", "sum", "wordwrap", "first", "list", "reverse",
+  "title", "xmlattr"
+];
+
+var JinjaCompletions = function() {
+
+};
+
+(function() {
+
+    this.getCompletions = function(keywordList, state, session, pos, prefix) {
+        var token = session.getTokenAt(pos.row, pos.column);
+
+        if (!token)
+            return [];
+
+        if (this.mayBeJinjaKeyword(token)) {
+            return this.getKeywordCompletions(keywordList, state, session, pos, prefix);
+        }
+
+        if (this.mayBeJinjaFilter(token)) {
+            return this.getFilterCompletions(state, session, pos, prefix);
+        }
+
+        if (this.mayBeJinjaVariable(token)) {
+            return this.getVariableCompletions(state, session, pos, prefix);
+        }
```

Comment:
Note to self: remove. Or better, still: detect variables of `for` loops or `set` directives.

But how?